niceman/distributions/tests/test_distribution.py

```diff
@@ -21,25 +21,48 @@
 
 def test_distributions(demo1_spec):
 
+    def mock_execute_command(command):
+        if type(command) is list:
+            if command == ['apt-cache', 'policy', 'libc6-dev:amd64']:
+                return (
+                    b'libc6-dev: \
+                        Installed: (none) \
+                        Candidate: 2.19-18+deb8u4 \
+                        Version table: \
+                            2.19-18+deb8u4 500 \
+                            500 http://archive.ubuntu.com/ubuntu xenial/universe amd64 Packages',
+                    0
+                )
+            if command == ['apt-cache', 'policy', 'afni']:
+                return (
+                    b'afni: \
+                        Installed: 16.2.07~dfsg.1-2~nd90+1 \
+                        Candidate: 16.2.07~dfsg.1-2~nd90+1 \
+                        Version table: \
+                            16.2.07~dfsg.1-2~nd90+1 500 \
+                            500 http://archive.ubuntu.com/ubuntu xenial/universe amd64 Packages',
+                    0
+                )
+        if type(command) is str:
+            if command.startswith('grep'):
+                return (None, 1)
+
     provenance = Provenance.factory(demo1_spec)
     distributions = provenance.get_distributions()
     distributions = items_to_dict(distributions)
     assert set(distributions.keys()) == {'conda', 'debian'}
     # Test DebianDistribution class.
     debian_distribution = distributions['debian']
     environment = MagicMock()
+    environment.execute_command = mock_execute_command
```

Comment:
nothing tests that apt file was generated/populated correctly?

Reply:
I added log checks to verify the right source string is added to /etc/apt/sources.list.d/niceman.sources.list